runner/webserver.ts

```diff
@@ -82,112 +72,136 @@ export function webserver(wct: Context): void {
-    urlPrefix = urlPrefix.replace('<basename>', path.basename(options.root));
-    options.webserver.webRunnerPath = urlPrefix + '/generated-index.html';
-
     // Hacky workaround for Firefox + Windows issue where FF screws up pathing.
     // Bug: https://github.com/Polymer/web-component-tester/issues/194
-
     options.suites = options.suites.map((cv) => cv.replace(/\\/g, '/'));
 
-    options.webserver.webRunnerContent = INDEX_TEMPLATE(options);
+    options.webserver._generatedIndexContent = INDEX_TEMPLATE(options);
   });
 
   wct.hook('prepare', async function() {
     const wsOptions = options.webserver;
+    const additionalRoutes = new Map<string, RequestHandler>();
+
+    let hasWarnedBrowserJs = false;
+    additionalRoutes.set('/browser.js', function(request, response) {
+      if (!hasWarnedBrowserJs) {
+        console.warn(`
 
-    const port = await getPort();
+          WARNING:
+          Loading WCT's browser.js from /browser.js is deprecated.
 
-    // `port` (and `webRunnerPath`) is read down the line by `BrowserRunner`.
-    wsOptions.port = port;
+          Instead load it from ../web-component-tester/browser.js
+          (or with the absolute url /components/web-component-tester/browser.js)
 
-    const app = express();
-    const server = http.createServer(app);
-    // `runTests` needs a reference to this (for the socket.io endpoint).
-    wct._httpServer = server;
+          Also be sure to \`bower install web-component-tester\`
 
-    // Debugging information for each request.
-    app.use(function(request, response, next) {
-      const msg = request.url + ' (' + request.header('referer') + ')';
-      wct.emit('log:debug', chalk.magenta(request.method), msg);
-      next();
+        `);
+        hasWarnedBrowserJs = true;
+      }
+      const browserJsPath = path.join(
+          options.root, 'bower_components', 'web-component-tester',
+          'browser.js');
+      send(request, browserJsPath).pipe(response);
     });
+    // TODO(rictic): detect that the user hasn't bower installed wct and die.
 
-    // Mapped static content (overriding files served at the root).
     _.each(wsOptions.staticContent, function(file, url) {
-      app.get(new RegExp(url), function(request, response) {
+      additionalRoutes.set(url, (request, response) => {
         response.set(DEFAULT_HEADERS);
         send(request, file).pipe(response);
       });
     });
 
-    // The generated web runner, if present.
-    if (wsOptions.webRunnerContent) {
-      app.get(wsOptions.webRunnerPath, function(request, response) {
-        response.set(DEFAULT_HEADERS);
-        response.send(wsOptions.webRunnerContent);
-      });
-    }
-
-    // At this point, we allow other plugins to hook and configure the
-    // webserver as they please.
-    await wct.emitHook('prepare:webserver', app);
+    const pathToGeneratedIndex =
+        `/components/${path.basename(options.root)}/generated-index.html`;
```

Comment:
We explicitly tell polyserve to use this packageName, so it won't infer it from the bower.json.

Refactoring this to serve somewhere else would be possible, but would involve touching a bunch of stuff and wouldn't actually solve the problem, as we'd need to know the path to the suites to load, which will be served out from the project under test, so we'd need to compute that path one way or another.

Clarified in the code a bit what's going on.